File: java/RabbitMQDemo/src/main/scala/codetosurvive1/helloworld/Send.scala
package codetosurvive1.helloworld

import com.rabbitmq.client.ConnectionFactory;

object Send {

  private val QUEUE_NAME="hello";

  def main(argv:Array[String]){

    val factory = new ConnectionFactory();
    factory.setHost("localhost");
    val connection = factory.newConnection();
    val channel = connection.createChannel();
    channel.queueDeclare(QUEUE_NAME,false,false,false,null);

    val message = "hello world!";
    channel.basicPublish("",QUEUE_NAME,null,message.getBytes("UTF-8"));
    println("sent message :"+message);

    channel.close();
    connection.close();
  }
}

File: java/RabbitMQDemo/src/main/java/io/github/codetosurvive1/helloworld/Send.java
package io.github.codetosurvive1.helloworld;

/**
 * Created by mac on 2017/1/7.
 */

import com.rabbitmq.client.ConnectionFactory;
import com.rabbitmq.client.Connection;
import com.rabbitmq.client.Channel;

public class Send {

    private final static String QUEUE_NAME = "hello";


    public static void main(String args[]) throws Exception{

        ConnectionFactory factory = new ConnectionFactory();
        factory.setHost("localhost");
        Connection connection = factory.newConnection();
        Channel channel = connection.createChannel();

        channel.queueDeclare(QUEUE_NAME,false,false,false,null);
        String message = "Hello world!";
        channel.basicPublish("",QUEUE_NAME,null,message.getBytes("UTF-8"));
        System.out.println(" [x] Sent '"+message+"'");

        channel.close();
        connection.close();

    }

}
